### PR TITLE
Dashboard draft

### PR DESCRIPTION
**Pull request by theZasa**
## Changes Proposed:
- First Draft of Dashboard
- To be discussed:
- Renaming of "Devices" -> "Connected Boards" 
- and "Types" -> "Board Types"

## Issues Addressed:
- None yet

## Tests Performed:
- Builds in Frontend

## How to Test the Changes:
- /packages/deployment-server-ui  yarn run dev 

## Known Issues and TODO List:
- [x] replace mock data with real data
- [x] fix padding of UI Elements
- [x] link dashboard cards to subpages [blocking, pages don't exist yet] 
- [x] Add more numbers to the top elements: 
- [x] "total deployments | successful Deployments | pending Deployments | terminated Deployments"
- [x] "connected devices | busy | free |" 
- [ ] Clean up DeploymentGraph (axis, format)

## Nice to have
- do fancy CSS things
- make UI elements as responsive as possible

## How to Test PRs
1) First fetch the Pull-Request from the main repo, replacing `xx` with the ID of the PR:\
`git fetch git@github.com:eclipsesource/cdtcloud-deploymentserver.git pull/xx/head:pr_xx`
2) Checkout the Pull-Request, again replacing `xx` with the PR-ID:\
`git checkout pr_xx`
3) Perform the necessary tests for the PR

### CODE REVIEW

**Comment by theZasa** [comment]
I suggest merging the current draft-state of the dashboard to use it as a baseline for further PRs. 
@WoH @Gozzim @KevinK-9

File: packages/deployment-server-ui/src/pages/Dashboard.tsx [hunk 1]
```diff
@@ -1,13 +1,62 @@
-import { Divider } from "antd";
 import RecentDeploymentList from "../components/Dashboard/RecentDeploymentList";
+import DeploymentsOverTimeGraph from "../components/Dashboard/DeploymentsOverTimeGraph";
 import defineFunctionalComponent from "../util/defineFunctionalComponent";
+import { Dashboard } from "deployment-server";
+import { useState, useEffect } from "react";
+import { useInterval } from "react-use";
+
+import { Card, Divider, Row, Col } from "antd";
+import "./DashboardCSS.css";
```
Comment (WoH) [inline packages/deployment-server-ui/src/pages/Dashboard.tsx:9]:
```suggestion
import "./Dashboard.css";
```

and rename the file

**Review by WoH — CHANGES_REQUESTED** [review]
Primarily blocking until the ico rights are clarified. The other things can be addressed in the meantime

File: packages/theia-extension/cdtcloud/src/browser/cdtcloud-widget/TypeSelect.tsx [hunk 2]
```diff
@@ -1,39 +0,0 @@
-import * as React from "react";
```
[the remark this answers is not in the document]
Comment (theZasa) [reply packages/theia-extension/cdtcloud/src/browser/cdtcloud-widget/TypeSelect.tsx:1]: This happend when i merged main into this branch

File: packages/deployment-server-ui/src/components/Dashboard/DeploymentsOverTimeGraph.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { useState, useEffect } from 'react';
+import { Line } from '@ant-design/plots';
+import defineFunctionalComponent from '../../util/defineFunctionalComponent';
+
+type DataEntries = {xField: string, yField: string}[]
+
+export default defineFunctionalComponent(function DeploymentsOverTimeGraph() {
```
[the remark this answers is not in the document]
Comment (theZasa) [reply packages/deployment-server-ui/src/components/Dashboard/DeploymentsOverTimeGraph.tsx:7]: Will be adressed in cleanup

**Comment by theZasa** [comment]
> Primarily blocking until the ico rights are clarified. The other things can be addressed in the meantime

Ico rights have been clarified. It is fine as it is under EPL

**Review by WoH — APPROVED** [review]

**Pull request merged**